### PR TITLE
State trie migration rococo runtime changes.

### PR DESCRIPTION
**Pull request by cheme**
This PR contains runtime change to switch state mine to a hybrid trie state, as needed for https://github.com/paritytech/devops/issues/1508.
Note that the account need to be change to the rights KEY_ROOT_MIG_CONTROLLER. CC @PierreBesson

### CODE REVIEW

File: runtime/rococo/src/lib.rs [hunk 1]
```diff
@@ -1375,6 +1375,9 @@ construct_runtime! {
 		// Gilts pallet.
 		Gilt: pallet_gilt::{Pallet, Call, Storage, Event<T>, Config} = 38,
 
+		// State trie migration pallet, only temporary.
+		StateTrieMigration: pallet_state_trie_migration = 41,
```
Comment (cheme) [inline runtime/rococo/src/lib.rs:1379]: This index was chosen a bit randomly, please tell me if there is a more suitable one (the crate will not be in the runtime forever).

File: runtime/rococo/src/lib.rs [hunk 2]
```diff
@@ -1463,57 +1466,103 @@ pub type Executive = frame_executive::Executive<
...
-		[pallet_xcm_benchmarks::fungible, pallet_xcm_benchmarks::fungible::Pallet::<Runtime>]
-		[pallet_xcm_benchmarks::generic, pallet_xcm_benchmarks::generic::Pallet::<Runtime>]
-	);
+			// Polkadot
+			// NOTE: Make sure to prefix these with `runtime_common::` so
+			// the that path resolves correctly in the generated file.
+			[runtime_common::auctions, Auctions]
+			[runtime_common::crowdloan, Crowdloan]
+			[runtime_common::claims, Claims]
+			[runtime_common::slots, Slots]
```
[the remark this answers is not in the document]
Comment (cheme) [reply runtime/rococo/src/lib.rs]: bad merge :see_no_evil:

File: runtime/rococo/src/lib.rs (hunk 1 above)
Comment (bkchr) [inline runtime/rococo/src/lib.rs:1379]:
```suggestion
		StateTrieMigration: pallet_state_trie_migration = 254,
```

And then move it down.

File: runtime/rococo/src/lib.rs [hunk 3]
```diff
@@ -1463,6 +1466,42 @@ pub type Executive = frame_executive::Executive<
...
+	pub const MigrationSignedDepositBase: Balance = 20 * CENTS * 100;
+	pub const MigrationMaxKeyLen: u32 = 512;
+}
+
+impl pallet_state_trie_migration::Config for Runtime {
+	type RuntimeEvent = RuntimeEvent;
+	type Currency = Balances;
+	type SignedDepositPerItem = MigrationSignedDepositPerItem;
+	type SignedDepositBase = MigrationSignedDepositBase;
+	// An origin that can control the whole pallet: should be Root, or a part of your council.
```
Comment (bkchr) [inline runtime/rococo/src/lib.rs:1481]:
```suggestion
```

File: runtime/rococo/src/lib.rs [hunk 4]
```diff
@@ -1463,6 +1466,42 @@ pub type Executive = frame_executive::Executive<
...
+#[test]
+fn ensure_key_ss58() {
+	use frame_support::traits::SortedMembers;
+	use sp_core::crypto::Ss58Codec;
+	let acc =
+		AccountId::from_ss58check("5DwBmEFPXRESyEam5SsQF1zbWSCn2kCjyLW51hJHXe9vW4xs").unwrap();
+	//panic!("{:x?}", acc);
+	assert_eq!(acc, MigController::sorted_members()[0]);
+}
+
```
Comment (bkchr) [inline runtime/rococo/src/lib.rs:1504]:
```suggestion
```

**Review by bkchr — APPROVED** [review]

**Comment by cheme** [comment]
Ping for review on this PR (@kianenigma since similar with statemine one but without the feature).

**Review by kianenigma — APPROVED** [review]

**Comment by cheme** [comment]
bot rebase

**Comment by cheme** [comment]
I am a bit hesitant to merge, I got all afternoon multiple failure at:
The assertion fails at block 10, I did some test and it constantly happen when I change `state_version` to 1 failing at https://github.com/paritytech/substrate/blob/38a955ba4a0c967659e6c944c56c664fce0f8f23/frame/babe/src/lib.rs#L378
the assertion fails at block 10 of collating_using_undying_collator.
Running locally I sometime have one of the peer passing it, and last ci run did work.

Is this test known to fail randomly (and maybe switching to state version 1 makes this more frequent)?

**Comment by bkchr** [comment]
@cheme which test is failing?

**Comment by cheme** [comment]
integration test `collating_using_undying_collator`, but last time I try locally it was passing, what I did see was :
```
2022-11-10 14:25:30 [//Alice] 🙌 Starting consensus session on top of parent 0x666ced27973721b50f6e08672486e61b2e802a40a34f115ad765cdeb29900dab    
2022-11-10 14:25:30 [//Alice] Evicting failed runtime instance error=Runtime panicked: assertion failed: pubkey.vrf_verify(transcript.clone(), vrf_output, vrf_proof).is_ok()
2022-11-10 14:25:30 [//Alice] Proposing failed: Import failed: Error at calling runtime api: Execution failed: Runtime panicked: assertion failed: pubkey.vrf_verify(transcript.clone(), vrf_output, vrf_proof).is_ok()    
2```
Looping.

**Comment by bkchr** [comment]
So you only have seen it failing in CI?

**Comment by cheme** [comment]
Well I did observe it locally too, but not anymore (I did apply this on substrate to check if state could be involve, but cant trigger it anymore:
```
+                                                       if !pubkey
+                                                               .vrf_verify(transcript.clone(), vrf_output, vrf_proof)
+                                                               .is_ok() {
+                                                               panic!(
+                                                                       "Fail at {:?}.",
+                                                                       (Self::randomness(), EpochIndex::<T>::get(), vrf_output, vrf_proof),
+                                                               );
+                                                       }
+/*
                                                        // NOTE: this is verified by the client when importing the block, before
                                                        // execution. we don't run the verification again here to avoid slowing
                                                        // down the runtime.
                                                        debug_assert!(pubkey
                                                                .vrf_verify(transcript.clone(), vrf_output, vrf_proof)
                                                                .is_ok());
-
+*/
```

that's a bit disconcerting

**Comment by bkchr** [comment]
Hmm yeah okay, you should really look into this to find out why this is happening.

**Comment by cheme** [comment]
Well I did trigger it three time on master also, but probably 3 out of 12 or even less frequent.
So maybe I was really not lucky this afternoon to observe it that often.

**Comment by bkchr** [comment]
So it also fails on master? Then let's merge this

**Comment by cheme** [comment]
bot merge

File: runtime/rococo/src/lib.rs [hunk 5]
```diff
@@ -108,14 +108,14 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
 	spec_name: create_runtime_str!("rococo"),
 	impl_name: create_runtime_str!("parity-rococo-v2.0"),
 	authoring_version: 0,
-	spec_version: 9290,
+	spec_version: 9291,
```
[the remark this answers is not in the document]
Comment (coderobe) [reply runtime/rococo/src/lib.rs]: (for future reference) no, these shouldn't be part of PRs

[the remark this answers is not in the document]
Comment (cheme) [reply runtime/rococo/src/lib.rs]: :+1: a mistake (what was merged to master do not include it).

**Pull request merged**